### ISSUE TITLE
New grep versions and warnings about stray /

### ISSUE THREAD

**Issue by matteocorti** [koalaman/shellcheck#2573]
#### For new checks and feature suggestions

According to POSIX regex standard:

```
"An ordinary character is a BRE that matches itself: any character in the
supported character set, except for the BRE special characters listed in
BRE Special Characters.
The interpretation of an ordinary character preceded by an unescaped
( '\' ) is undefined, except for: [...]"
```

grep is giving a warning since 3.8:

```
$ grep -V
grep (GNU grep) 3.8
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Mike Haertel and others; see
<https://git.sv.gnu.org/cgit/grep.git/tree/AUTHORS>.

$ grep ' ' check_ssl_cert > /dev/null 
$ grep '\ ' check_ssl_cert > /dev/null 
ggrep: warning: stray \ before white space
$ grep '[ ]' check_ssl_cert > /dev/null 
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

Should warn about the usage of stray \

**Comment by noloader** [comment]
Also see [dfa: new option DFA_STRAY_BACKSLASH_WARN](https://lists.gnu.org/archive/html/grep-devel/2022-06/msg00006.html) on the grep-devel mailing list.

**Comment by ale5000-git** [comment]
Hi,
any update on this?

Another example:
```
$ echo ' href="123" ' | grep -o -e ' href=\"[^"]*\" '
grep: warning: stray \ before "
grep: warning: stray \ before "
```

**Comment by wileyhy** [comment]
OP's code:

  <sh>
  echo ' href="123" ' \
    | grep -o -e ' href=\"[^"]*\" '
  </sh>

I don't know enough about Haskell to know whether this level of granularity
is possible, but whether it's POSIX-2024 `grep`, or GNU `grep`, the operand
to '-e' appears to me to be a malformed BRE.

What happens when a script tries to `eval` a `grep` command with this kind
of RE syntax, or when word splitting is intended, etc. -- when a level of
quoting is intended to be reduced?
